tests/kernel/semaphore/sys_sem/src/main.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <ztest_error_hook.h>
+
+/* Macro declarations */
+#define STACK_SIZE (512 + CONFIG_TEST_EXTRA_STACKSIZE)
+#define SEM_INIT_VAL (0U)
+#define SEM_MAX_VAL  (3U)
+#define TOTAL_MAX (4U)
+#define PRIO 5
+#define LOW_PRIO 8
+#define HIGH_PRIO 2
+
+static K_THREAD_STACK_DEFINE(stack1, STACK_SIZE);
+static K_THREAD_STACK_DEFINE(stack2, STACK_SIZE);
+static K_THREAD_STACK_DEFINE(stack3, STACK_SIZE);
+static K_THREAD_STACK_DEFINE(stack4, STACK_SIZE);
+static K_THREAD_STACK_ARRAY_DEFINE(multi_stack_give, TOTAL_MAX, STACK_SIZE);
+static K_THREAD_STACK_ARRAY_DEFINE(multi_stack_take, TOTAL_MAX, STACK_SIZE);
```

Comment:
Surely all these stacks aren't being used simultaneously, right?  Maybe unify them?  I guess it's true that this is a userspace test and userspace-enabled devices tend to be bigger, but traditionally our test suite has been expected to run on minimal hardware.

Reply:
Thanks for comments, good knowledge, I will unify them.